Feature/feature search

cs-daniel-lima opened the pull request:
Seguindo as recomendações do @cs-joao-souza aproveitei o início desta feature para refatorar algumas coisas que considero importantes para seguir em frente com a feature e seria bom já ter isso na develop para que todos possam se beneficiar.

**Principais mudanças:**
- Gemfile (bundler) garantindo versão do cocoapods
- Garantidas versões de libs no Podfile e Cartfile
- .gitattributes removido
- Todas as imagens adicionadas ao assets catalog
- Algumas classes já se encontram refatoradas seguindo as recomendações enviadas via slack (CardListView, MgicTabBarController e MagicSearchBar)

cs-daniel-lima committed:
Guarantees the version of libs and gems
 Set the appropriate version of libs  in the Podfile file and the Cartfile, and also set the version of the cocoapods in Gemfile

cs-daniel-lima committed:
remove .gitattributes

cs-daniel-lima committed:
Remove image files that were not on the assets catalog
 buscar.png and fundo.png added to assets catalog and and xcscheme identification is now fixed to 'magicbyirineu' instead of 'magicbyirineu 2'

cs-daniel-lima committed:
Refactor the CardListView class, remove FavoriteView, and update CodeView protocol
 remove unnecessary lazy statements and change vars to let where is appropriate to do so
 delete empty class FavoriteView
 rearrange the order in which the CodeView protocol methods are called

cs-daniel-lima committed:
Refactor MagicTabBarController and MagicSearchBar
The MagicTabBarController and MagicSearchBar are refactored here to reduce code smell and add dependency injection

cs-daniel-lima committed:
Update MagicTabBarMock and fix problem detected on a snapshot from the MagicSearchBar
MagicSearchBar now call the methods setupTextFieldBorders and setupTextFieldPlaceholder on the willMove method

cs-joao-souza commented on magicbyirineu/Modules/CardList/View/CardListViewController.swift:
```diff
@@ -30,7 +39,7 @@ class CardListViewController: UIViewController {
     }
     
     override func viewDidAppear(_ animated: Bool) {
-        screen.searchBar.setupSearchBar()
```
Ainda falta chamar o super aqui.

cs-daniel-lima replied:
Correção submetida

cs-joao-souza commented on magicbyirineu/Modules/CardList/View/MagicSearchBar.swift:
```diff
@@ -36,16 +60,21 @@ class MagicSearchBar: UISearchBar {
-    func setupSearchBar(){
+    func afterDisplaySetup(){
         
         let textField = value(forKey: "_searchField") as! UITextField
```
Perigoso isso aqui. Por que não criaram seu próprio componente de busca?

cs-daniel-lima replied:
Correção submetida

cs-joao-souza commented on magicbyirineu/Modules/Favorites/View/FavoritesViewController.swift:
```diff
@@ -20,26 +29,14 @@ class FavoritesViewController: UIViewController {
-        // Do any additional setup after loading the view.
     }
     
     override func viewDidAppear(_ animated: Bool) {
```
Falta chamar o super.

cs-daniel-lima replied:
Correção submetida

cs-daniel-lima committed:
Refactoring
Calling some missing super and making sure to avoid force unwrap in MagicSearchBar

cs-joao-souza commented on magicbyirineu/Modules/CardList/View/CardListViewController.swift:
```diff
@@ -34,12 +34,11 @@ class CardListViewController: UIViewController {
     
     override func viewDidAppear(_ animated: Bool) {
-        screen.searchBar.afterDisplaySetup()
+        screen.searchBar.additionalSetupAfterDisplay()
```
o super é chamado sempre antes.

cs-daniel-lima committed:
Super called properly
